SMLHelper/Options/ModOptionAttribute.cs

```diff
@@ -0,0 +1,50 @@
+﻿namespace SMLHelper.V2.Options
+{
+    using Handlers;
+    using Interfaces;
+    using Json;
+    using System;
+
+    /// <summary>
+    /// Abstract base attribute used to signify the attributed public member should generate a <see cref="ModOption"/>
+    /// in a mod's options menu.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Field | AttributeTargets.Property | AttributeTargets.Method, AllowMultiple = false)]
+    public abstract class ModOptionAttribute : Attribute, IModOptionAttribute
+    {
+        /// <summary>
+        /// The label to use when displaying the field in the mod's options menu.
+        /// </summary>
+        public string Label { get; set; }
+
+        /// <summary>
+        /// The Id to be used for the field in the mod's option menu. If none is specified, one will be automatically generated when
+        /// your <see cref="ConfigFile"/> is registered to the <see cref="OptionsPanelHandler"/>. This means it will
+        /// change every time the game is launched, but is guaranteed to be unique. If you would like to specify an Id to use for
+        /// internal comparisons, you can do so here.
+        /// </summary>
+        public string Id { get; set; } = Guid.NewGuid().ToString().Replace('-', '_');
+
+        /// <summary>
+        /// The order in which to display fields in the mod's option menu, in ascending order. If none is specified, the order will be
+        /// automatically set.
+        /// </summary>
+        public int Order { get; set; } = i++;
+        private static int i = 0;
```

Comment:
I looked at this already and decided that:

1. since we upgrading to .NET Standard soon because of changes to Subnautica and Below Zero, hacking in support for features not available in .NET Framework 4.0 is pointless to do now

1. As mentioned in the resolved conversation, `[CallerLineNumber]` is a **compile time** feature. It is *not* available at *runtime*. This means that it *cannot* work across assemblies. A mod using the attribute will live in another assembly to SMLHelper, and therefore this class will have no way to access the *line numbers* of the *original file*, which result in all 0s, as seen in this example:
![image](https://user-images.githubusercontent.com/45315526/89323095-23512d00-d67d-11ea-82f6-971453a6ce27.png)

Reply:
Actually, for point 2, I was mistaken, and I must apologise - my testing methodology was imprecise and this can actually work.

I've come around on this - but I'd still rather wait until we officially move to .NET Standard to implement, as if we implement it before then, we'll have a namespace conflict.

Reply:
Yeah, that line parameter should be in the derived attribute. And yeah, considering that exp branch uses 4.6.2, it's better to wait for update, but no one knows when it'll be.

Reply:
.NET Standard now actually, not even just .NET 4.6.2.

Reply:
I don't mind waiting for it to come to stable for the ordering to be perfect. It's better that than conflicting with the namespace so mods have yet another reason to be broken when the update comes.

Reply:
It'll probably be more compiler warnings about conflicting types than actual crashes, but yeah, it's better to not add new hacks anyway.